### DISCUSSION TITLE
Qtile. Fedora. Pip.

### DISCUSSION THREAD

**Discussion by HighTechTatoka** [qtile/qtile#4801] [Q&A]
Hello, dear Qtile community! I want to ask you some questions:
1. Is Qtile only for Arch Linux? Will you support Qtile for Fedora via dnf? If not, why?
2. Can I use all Pythons’s possibilities in Qtile? I mean, can I integrate some programms in Qtile, which were made by Python. Or Qtile supports only limited Python's possibilities?
3. If I install qtile via pip, will I get errors with Dnf package dependencies?

**Comment by elParaguayo** [comment]
1. Qtile is not just for Arch, it should work fine on Fedora. However, we are not responsible for packaging it. We rely on distro maintainers/users, to package qtile for that distro.
2. Can you explain what you mean by "integrate"
3. I would expect that Fedora won't let you install packages via pip as system packages. There should be nothing stopping you using pip to install locally/inside a virtual environment.

**Reply by HighTechTatoka** [reply]
I mean, can I add some functionality in Qtile, such as neural network, for example.
Why Do Fedora not allow me to install Qtile via Pip? Fedora does not have qtile package, and I think to install qtile from source is only single opportunity for Fedora.

**Reply by elParaguayo** [reply]
> I mean, can I add some functionality in Qtile, such as neural network, for example.

Being in written in python means you can have qtile however you want.

> Why Do Fedora not allow me to install Qtile via Pip? 

If you install system packages via pip then you could overwrite files installed by another package but your package manager wouldn't know anything about these files. So, it's a really bad idea to do that.

>Fedora does not have qtile package, and I think to install qtile from source is only single opportunity for Fedora.

That's fine and it should be perfectly possible to do.

Might also be worth looking at this: https://copr.fedorainfracloud.org/coprs/frostyx/qtile-f39/

**Reply by HighTechTatoka** [reply]
Installing from source is installing via pip. Is it right?

**Reply by HighTechTatoka** [reply]
And I have another one question. If "pip install qtile" is dangerous command, why was it written in Docs? I have only one description about that - it is only for Arch and Arch based distros, because they have rolling release. But anyway...

**Reply by elParaguayo** [reply]
It's not a dangerous command on its own. It is dangerous if you try to use it to install system wide packages (which would typically require `sudo`). 

This isn't just a rolling release issue. Any distro with a package manager has the same issue.

**Reply by HighTechTatoka** [reply]
Fedora have added Qtile in officical repository recently. And Fedora's users can install that via DNF. Thanks for help.
